### PR TITLE
Fixes #1617 - Genetics can only reclaim active genes

### LINKED ISSUE #1617: Not possible to reclaim weakener-spliced mutations.
**Describe the bug**
Weakener is a chromosome that aims to increase the material gained on reclaiming a mutation that has it spliced.
However, to splice the chromosome you need to activate the mutation and you can't reclaim activated mutations. The chromosome's description also claims it's supposed to make the mutation reclaimable, but it does not result in such.
Unless I'm missing something that makes it useless, with it already not seeing much use.

**To Reproduce**
1. Activate a mutation.
2. Splice it with the weakener chromosome, which claims "Makes a gene reclaimable and doubles the amount of materials you get from reclaiming it.".
3. The gene cannot be reclaimed.

**Expected behavior**
Being able to reclaim the mutation as well as getting extra material for it.

### PR DESCRIPTION
**Pull request by amylizzle**
## About the PR 
Fixes #1617 by making reclaim only work on _active_ mutations, which makes a lot more sense to my mind anyway.

## Why's this needed? 
You could previously only reclaim inactive mutations, and you could only splice active mutations, making the weakener chromosome pointless.

## Changelog

```changelog
(u)Amylizzle
(+)Genetics reclaimer can now only reclaim active genes.
```

### CODE REVIEW

File: tgui/packages/tgui/interfaces/GeneTek/BioEffect.js [hunk 1]
```diff
@@ -205,7 +205,7 @@ export const BioEffect = (props, context) => {
             Check Stability
           </Button>
         )}
-        {haveDevice(equipmentCooldown, "Reclaimer") && isPotential && !!canReclaim && (
+        {haveDevice(equipmentCooldown, "Reclaimer") && (isStorage || isActive) && !!canReclaim && (
```
Comment (pgmzeta) [inline tgui/packages/tgui/interfaces/GeneTek/BioEffect.js:208]: should this use `activeOrStorage`

**Comment by ThatFiveGuys** [comment]
Why not allow them to reclaim both inactive and active mutations? Seems the best for qol

File: tgui/packages/tgui/interfaces/GeneTek/BioEffect.js (hunk 1 above)
Comment (amylizzle) [reply tgui/packages/tgui/interfaces/GeneTek/BioEffect.js:208]: yes it should.

**Comment by amylizzle** [comment]
> Why not allow them to reclaim both inactive and active mutations? Seems the best for qol

Mostly it seems like a thing that should be gated behind the gene being active, otherwise you just randomise genes and spam reclaim for free materials.

**Pull request merged**